### PR TITLE
[Show | Command Reference] Add Port breakout Show Command

### PR DESCRIPTION
**Pull request by samaity**
Signed-off-by: Sangita Maity <sangitamaity0211@gmail.com>

**- What I did**
Implemented show interface breakout sub-command to show the port capability and the current breakout mode.

Available commands are mentioned below.
```
show interfaces breakout --help
show interfaces breakout current-mode
```
**- How I did it**
using platform.json, hwsku.json and config-db BREAKOUT_CFG table.

**- How to verify it**
```
admin@lnos-x1-a-fab01:~$ show interfaces breakout --help
Usage: show interfaces breakout [OPTIONS] COMMAND [ARGS]...

  Show interface breakout

Options:
  -?, -h, --help  Show this message and exit.

Commands:
  current-mode  Show interface breakout current-mode
```
```
admin@lnos-x1-a-fab01:~$ show interfaces breakout
{
    "Ethernet0": {
        "index": "1,1,1,1",
        "default_brkout_mode": "1x100G[40G]",
        "child ports": "Ethernet0",
        "child port speed": "100G",
        "breakout_modes": "1x100G[40G],2x50G,4x25G[10G]",
        "Current Breakout Mode": "1x100G[40G]",
        "lanes": "65,66,67,68",
        "alias_at_lanes": "Eth1/1, Eth1/2, Eth1/3, Eth1/4"
    },
    "Ethernet4": {
        "index": "2,2,2,2",
        "default_brkout_mode": "1x100G[40G]",
        "child ports": "Ethernet4,Ethernet5,Ethernet6,Ethernet7",
        "child port speed": "25G,10G,25G,25G",
        "breakout_modes": "1x100G[40G],2x50G,4x25G[10G]",
        "Current Breakout Mode": "4x25G",
        "lanes": "69,70,71,72",
        "alias_at_lanes": "Eth2/1, Eth2/2, Eth2/3, Eth2/4"
    },
    "Ethernet8": {
        "index": "3,3,3,3",
        "default_brkout_mode": "1x100G[40G]",
        "child ports": "Ethernet8",
        "child port speed": "100G",
        "breakout_modes": "1x100G[40G],2x50G,4x25G[10G]",
        "Current Breakout Mode": "1x100G[40G]",
        "lanes": "73,74,75,76",
        "alias_at_lanes": "Eth3/1, Eth3/2, Eth3/3, Eth3/4"
   },... continue
}
```
```
admin@lnos-x1-a-fab01:~$ show interfaces breakout current-mode Ethernet0
+-------------+-------------------------+
| Interface   | Current Breakout Mode   |
+=============+=========================+
| Ethernet0   | 4x25G[10G]              |
+-------------+-------------------------+
```
```
admin@lnos-x1-a-fab01:~$ show interfaces breakout current-mode
+-------------+-------------------------+
| Interface   | Current Breakout Mode   |
+=============+=========================+
| Ethernet0   | 4x25G[10G]              |
+-------------+-------------------------+
| Ethernet4   | 4x25G[10G]              |
+-------------+-------------------------+
| Ethernet8   | 4x25G[10G]              |
+-------------+-------------------------+
```

### CODE REVIEW

**Comment by vasant17** [comment]
As discussed, lets add two more fields, breakout_interfaces, and their speeds. For example:
```
 "Ethernet0": {
        "index": "1,1,1,1",
        "default_brkout_mode": "4x25G[10G]",
        "lanes": "65,66,67,68",
        "brkout_mode": "4x25G[10G]",
        "breakout_modes": "1x100G[40G],2x50G,4x25G[10G]",
        "alias_at_lanes": "Eth1/1, Eth1/2, Eth1/3, Eth1/4"
        "breakout_interfaces": "Ethernet0, Ethernet1, Ethernet2, Ethernet3",
        "breakout_speeds": "25G, 25G, 25G, 25G"
    },
```

**Comment by samaity** [comment]
> As discussed, lets add two more fields, breakout_interfaces, and their speeds. For example:
> 
> ```
>  "Ethernet0": {
>         "index": "1,1,1,1",
>         "default_brkout_mode": "4x25G[10G]",
>         "lanes": "65,66,67,68",
>         "brkout_mode": "4x25G[10G]",
>         "breakout_modes": "1x100G[40G],2x50G,4x25G[10G]",
>         "alias_at_lanes": "Eth1/1, Eth1/2, Eth1/3, Eth1/4"
>         "breakout_interfaces": "Ethernet0, Ethernet1, Ethernet2, Ethernet3",
>         "breakout_speeds": "25G, 25G, 25G, 25G"
>     },
> ```

Yes, updated the PR.

File: show/main.py [hunk 1]
```diff
@@ -21,8 +21,15 @@
 from swsssdk import SonicV2Connector
 
 import mlnx
+from collections import OrderedDict
```
Comment (jleveque) [inline show/main.py:24]: This is a system import move above the line, `from pkg_resources import parse_version` (line12) to follow PEP8 standards:

> Imports should be grouped in the following order:
>
> 1. Standard library imports.
> 2. Related third party imports.
> 3. Local application/library specific imports.
>
> You should put a blank line between each group of imports.

https://www.python.org/dev/peps/pep-0008/#imports

File: show/main.py [hunk 2]
```diff
@@ -21,8 +21,15 @@
...
+from portconfig import get_child_ports
```
Comment (jleveque) [inline show/main.py:25]: This is a SONiC library import. Move above the line, `from swsssdk import ConfigDBConnector` (line 20)

File: show/main.py [hunk 3]
```diff
@@ -797,6 +813,102 @@ def alias(interfacename):
...
     click.echo(tabulate(body, header))
 
+
+#
+# 'breakout' group ###
+#
+@interfaces.group(invoke_without_command=True)
+@click.pass_context
+def breakout(ctx):
+    """Show interface breakout"""
```
Comment (jleveque) [inline show/main.py:823]: This docstring becomes the help message. It should be a little more descriptive. Even if simply, `Show interface breakout mode info`

File: show/main.py [hunk 4]
```diff
@@ -797,6 +813,102 @@ def alias(interfacename):
...
+
+        for port_name in platformDict.keys():
+            curBrkout_mode = curBrkout_tbl[port_name]["brkout_mode"]
+
+            # Update deafult breakout mode and current breakout mode to platformDict
+            platformDict[port_name].update(hwskuDict[port_name])
+            platformDict[port_name]["Current Breakout Mode"] = curBrkout_mode
+
+            # List all the child ports if present
+            child_portDict = get_child_ports(port_name,curBrkout_mode, platformFile)
```
Comment (jleveque) [inline show/main.py:860]: Add space after first comma

File: show/main.py [hunk 5]
```diff
@@ -797,6 +813,102 @@ def alias(interfacename):
...
+            if not child_portDict:
+                click.echo("Can not find ports from {} file ".format(PLATFORM_JSON))
```
Comment (jleveque) [inline show/main.py:862]: s/Can not/Cannot/

File: show/main.py (hunk 1 above)
Comment (samaity) [reply show/main.py]: #done

File: show/main.py (hunk 4 above)
Comment (samaity) [reply show/main.py]: #done

File: show/main.py (hunk 3 above)
Comment (samaity) [reply show/main.py]: #done

File: show/main.py (hunk 5 above)
Comment (samaity) [reply show/main.py]: #done

File: show/main.py (hunk 2 above)
Comment (samaity) [reply show/main.py]: #done

**Comment by jleveque** [comment]
Please fix LGTM alert

**Comment by samaity** [comment]
> Please fix LGTM alert

#done

**Comment by jleveque** [comment]
Retest this please

**Comment by jleveque** [comment]
Retest this please

**Comment by zhenggen-xu** [comment]
Retest this please

**Comment by daall** [comment]
retest this please

**Comment by zhenggen-xu** [comment]
retest this please

**Pull request merged**